src/utils/katakana_to_hiragana.rs

```diff
@@ -93,6 +103,10 @@ pub(crate) fn katakana_to_hiragana_with_opt(input: &str, is_destination_romaji:
 
             hira.push(hira_char);
             previous_kana = Some(hira_char);
+        } else if is_char_halfwidth_katakana(input_char) {
+            let result = HALFWIDTH_KATAKANA_TO_HIRAGANA_NODE_TREE.get(&chars[index..]);
```

Comment:
I'd prefer a `pos` variable that gets incremented instead of the `previous_read_forward_count`.